src/cudamatrix/cu-matrix-test.cc

```diff
@@ -1111,6 +1111,26 @@ template<typename Real> static void UnitTestCuMatrixAddMatMatElements() {
   KALDI_ASSERT(M.Sum() != 0.0);
 }
 
+template<typename Real> static void UnitTestCuMatrixAddMatMatDivMat() {
+  // M = a * b / c (by element; when c = 0, M = a)
+  MatrixIndexT dimM = 100 + Rand() % 255, dimN = 100 + Rand() % 255;
+  CuMatrix<Real> M(dimM, dimN), A(dimM, dimN), B(dimM, dimN), C(dimM, dimN);
+  CuMatrix<Real> ref(dimM, dimN);
+  M.SetRandn();
+  A.SetRandn();
+  B.SetRandn();
+  C.SetRandn();
+
+  M.AddMatMatDivMat(A,B,C);
+  ref.AddMatMatElements(1.0, A, B, 0.0);
```

Comment:
There is a slight theoretical issue here, that the function AddMatMatDivMat has special behavior if an element of c is exactly 0.  The prob. that this happens is very small, of course.  we could just ignore it, but put a comment...